### ISSUE TITLE
Triple vvv and above hangs rsync

### ISSUE THREAD

**Issue by dr-who** [WayneD/rsync#231]
Hi there,

We have had a case where the addition of an extra v (rsync -arvvv ... ...) cause rsync to hang. There are reports from https://bugs.launchpad.net/ubuntu/+source/rsync/+bug/1528921 and comments such as "I can only manually reproduce it when executing my rsync command with tripe verbosity:", so it's not just us.

Looking at options.c, the 6 values are defined for char *debug_verbosity[] but only three are defined for char *info_verbosity[].

What happens when three v are set? I note that adding 7 values to char *info_verbosity array causes a compilation error of course, but not with less than 6. But what happens with the redirection with three v values, when elements 3,4,5 are not defined?

**Comment by WayneD** [comment]
Did you test the latest git source?  Why are you running -vvv?  For a lot of verbosity, it's good to use `--stderr=all` (aka `--msgs2stderr`).

**Comment by dr-who** [comment]
yes, this is the very latest source. Probably like most of the users we just wanted to know exactly what was going on, it was easier than adding an strace. We have >10TB file systems and large files so it's nice to watch.

**Comment by WayneD** [comment]
There's nothing wrong with the info_verbosity setup -- it has a defined size to make it the same as the debug_verbosity array, so the unmentioned initializers are all NULL (since it's static).

**Comment by WayneD** [comment]
The uninitialized error is a bit weird.  It points to: `if (iobuf.in_fd >= 0 && FD_ISSET(iobuf.in_fd, &r_fds)) {` which I thought shouldn't have any uninitialized memory because `FD_CLEAR(&r_fds)` was called earlier, but maybe iobuf.in is greater than the default FD_SETSIZE?  It may only be 256 on some systems.  I'll check into that.  That could be a significant issue.

**Comment by dr-who** [comment]
It looked to me with multiple v's the INFO fields were added to the  info_words[] structure until it was larger than COUNT_INFO+1. I'll keep hunting. 

And yes, the uninitialized error did seem like a worry.

**Comment by WayneD** [comment]
I'll note that fixing this is not a very high priority given the better `-ii` option choice & `--msgs2stderr`.  Especially since the only things that can really be done in the current protocol is to either throw more memory at the backlog of messages and/or to slow down the transfer by not doing so much checksum work in the pipeline while waiting for the sender to answer back.
